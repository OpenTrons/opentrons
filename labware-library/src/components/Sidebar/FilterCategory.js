// @flow
// filter labware by category
import * as React from 'react'
import { Link } from 'react-router-dom'
import cx from 'classnames'

import { getAllCategories, buildFiltersUrl } from '../../filters'
import styles from './styles.css'

import type { FilterParams } from '../../types'

// TODO(mc, 2019-03-18): i18n, duplicated in
//   labware-library/src/components/LabwareList/LabwareCard.js
const EN_CATEGORY_LABELS = {
  all: 'All',
  tubeRack: 'Tube Rack',
  tipRack: 'Tip Rack',
  wellPlate: 'Well Plate',
  trough: 'Trough',
  trash: 'Trash',
  other: 'Other',
}

export type FilterCategoryProps = {
  filters: FilterParams,
}

export default function FilterCategory(props: FilterCategoryProps) {
  const { filters } = props
  const categories = getAllCategories()

  return (
    <ul className={styles.filter_category}>
      {categories.map(c => (
        <li key={c} className={styles.filter_category_item}>
          <Link
            to={buildFiltersUrl({ ...filters, category: c })}
            className={cx(styles.filter_category_link, {
              [styles.selected]: c === filters.category,
            })}
          >
            {EN_CATEGORY_LABELS[c]}
          </Link>
        </li>
      ))}
    </ul>
  )
}
