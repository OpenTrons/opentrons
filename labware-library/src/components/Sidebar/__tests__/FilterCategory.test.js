// @flow
// FilterCategory component tests
import * as React from 'react'
import Renderer from 'react-test-renderer'

import FilterCategory from '../FilterCategory'

describe('FilterCategory', () => {
  test('component renders', () => {
    const tree = Renderer.create(<FilterCategory />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  // TODO(mc, 2019-03-14): test category list population
})
