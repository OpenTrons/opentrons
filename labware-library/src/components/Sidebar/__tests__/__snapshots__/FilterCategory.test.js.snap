// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterCategory component renders 1`] = `
<ul
  className="filter_category"
>
  <li
    className="filter_category_item"
    key="all"
  >
    <Link
      className="filter_category_link selected"
      to="/?"
    >
      All
    </Link>
  </li>
  <li
    className="filter_category_item"
    key="tubeRack"
  >
    <Link
      className="filter_category_link"
      to="/?category=tubeRack"
    >
      Tube Rack
    </Link>
  </li>
  <li
    className="filter_category_item"
    key="wellPlate"
  >
    <Link
      className="filter_category_link"
      to="/?category=wellPlate"
    >
      Well Plate
    </Link>
  </li>
</ul>
`;
